parlai/agents/transformer/transformer.py

```diff
@@ -117,17 +115,31 @@ def build_model(self, states=None):
             )
         return self.model
 
+    def batchify(self, obs_batch, sort=False,
+                 is_valid=lambda obs: 'text_vec' in obs or 'image' in obs):
+        """Override so that we can add memories to the Batch object."""
+        batch = super().batchify(obs_batch, sort, is_valid)
+        if self.opt['use_memories']:
+            valid_obs = [(i, ex) for i, ex in enumerate(obs_batch) if is_valid(ex)]
+            valid_inds, exs = zip(*valid_obs)
+            mems = None
+            if any('memory_vecs' in ex for ex in exs):
+                mems = [ex.get('memory_vecs', None) for ex in exs]
+            batch.memory_vecs = mems
+        return batch
+
+    def _vectorize_memories(self, obs):
+        raise NotImplementedError(
+            'Abstract class: user must implement this function to use memories'
```

Comment:
Will this change? I don't think that Transformer is an abstract class...

Reply:
discussed offline, Transformer is not an abstract class, but in order to use the memnet aspect we will need to subclass it. the reason i made this method abstract was because the way we use memories is completely task dependent

Reply:
I agree with Emily's logic.

Should this actually be in torch ranker agent.

If it stays in  transformer, imho we should leave a more descriptive explanation here.

Reply:
Also does this necessitate fixing MemNet? Not sure we have a good test for it yet.

Reply:
mems will always be None so it won't break anything

Reply:
as far as moving to torch ranker agent, seems reasonable?

Reply:
Let's just add a comment TODO: bubble this up to TRA? and unblock this PR